src/hooks/useApiThrottle.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+
+type ApiCall = (...args: any[]) => Promise<any>;
+
+interface UseApiThrottleProps<T extends ApiCall> {
+  fn: T;
+  callback?: (res: Awaited<ReturnType<T>>) => void;
+}
+
+/**
+ * Prevent additional API call until the most recent one has completed.
+ */
+const useApiThrottle = <T extends ApiCall>(props: UseApiThrottleProps<T>) => {
+  const [fetching, setFetching] = React.useState(false);
+
+  const fn: (...args: Parameters<T>) => Promise<void> = React.useCallback(
+    async (...args) => {
+      if (fetching) {
+        return;
+      }
+      setFetching(true);
+      await props.fn(...args).then(props.callback);
```

Comment:
can `callback` be combined into `fn`? that way you're just passing in a single function into `useApiThrottle`

Reply:
Discussed with @johnny-t06. We think having `callback` separately make it nice to use the hooks directly using a client-side API without the need to wrap it.